docs/content/sips/009-auditing-third-party-dependencies.md

```diff
@@ -0,0 +1,149 @@
+from [Wasmtime](https://github.com/bytecodealliance/wasmtime/blob/main/supply-chain/audits.toml)
+and [Mozilla](https://hg.mozilla.org/mozilla-central/raw-file/tip/supply-chain/audits.toml).
+
+### What is the initial state of the `supply-chain` directory?
+
+Initially, all dependencies that are not already audited by a trusted organization
+will be added as a exemptions - which effectively marks the starting point of the
+auditing process as trusted. This is to avoid the potentially massive initial effort
+required to validate _all_ transitive dependencies. However, the first time a dependency
+would get updated, `cargo vet` will require explicitly auditing the new version,
+or manually adding it as an exemption.
+
+### How is an audit performed?
+
+[Performing audits](https://mozilla.github.io/cargo-vet/performing-audits.html)
+can be the result of two events:
+
+- manually removing a dependency from the exemptions list
+- a dependency has been updated
+
+When that happens, `cargo vet` will fail:
+
+```bash
+$ cargo vet
+  Vetting Failed!
+
+  3 unvetted dependencies:
+      bar:1.5 missing ["safe-to-deploy"]
+      baz:1.3 missing ["safe-to-deploy"]
+      foo:1.2.1 missing ["safe-to-deploy"]
+
+  recommended audits for safe-to-deploy:
+      cargo vet diff foo 1.2 1.2.1  (10 lines)
+      cargo vet diff bar 2.1.1 1.5  (253 lines)
+      cargo vet inspect baz 1.3     (2033 lines)
+
+  estimated audit backlog: 2296 lines
+
+  Use |cargo vet certify| to record the audits.
+```
+
+[From the documentation](https://mozilla.github.io/cargo-vet/performing-audits.html):
+
+> Note that if other versions of a given crate have already been verified,
+there will be multiple ways to perform the review: either from scratch, or
+relative to one or more already-audited versions. In these cases, `cargo vet`
+computes all the possible approaches and selects the smallest one.
+>
+> You can, of course, choose to add one or more unvetted dependencies to the
+exemptions list instead of auditing them. This may be expedient in some situations,
+though doing so frequently undermines the value provided by the tool.
+
+Once a package is chosen for an audit, `cargo vet inspect` guides the user
+through performing the audit:
+
+```bash
+$ cargo vet inspect baz 1.3
+You are about to inspect version 1.3 of 'baz', likely to certify it for "safe-to-deploy", which means:
+   ...
+You can inspect the crate here: https://sourcegraph.com/crates/baz@v1.3
+
+(press ENTER to open in your browser, or re-run with --mode=local)
+
+$ cargo vet certify baz 1.3
+
+  I, Alice, certify that I have audited version 1.3 of baz in accordance with
+  the following criteria:
+
+  ...
+
+ (type "yes" to certify): yes
+
+  Recorded full audit of baz version 1.3
+```
+
+Similarly, `cargo vet diff` will help guide the user through reviewing the diff
+between two versions and certifying the audit.
+
+Finally, `cargo vet suggest` is the command whose goal is to guide maintainers through
```

Comment:
Would we want to run this command on a regular cadence (perhaps similar to -- or included in -- the `Run Rust audits` GH action)?

This might tie into an approach to reduce the amount of exemptions (apart from dependency upgrades) over time.

Reply:
That sounds like a good idea — I'd like for us to get a bit of experience with this setup first before introducing this — but I really like the idea!